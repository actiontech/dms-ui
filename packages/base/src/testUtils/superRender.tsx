import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import { renderHook } from '@testing-library/react-hooks';
import lightTheme from '@actiontech/shared/lib/theme/light';
import basePackageTheme from '../theme/light';

const themeData = {
  ...lightTheme,
  ...basePackageTheme
};

export const baseSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData
  });
};

export const baseSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData
  });
};
