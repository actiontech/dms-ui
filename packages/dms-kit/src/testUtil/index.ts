export * from './superRender';
export * from './customQuery';
export * from './common';
