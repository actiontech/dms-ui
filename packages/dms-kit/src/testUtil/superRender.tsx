import { render, renderHook as renderHookReact } from '@testing-library/react';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { ConfigProvider, theme as antdTheme } from 'antd';
import zhCN from 'antd/locale/zh_CN';
import { ThemeProvider } from '@mui/system';
import { StyledEngineProvider } from '@mui/material/styles';
import lightTheme from '../theme/light';
import { ToolkitStore } from '@reduxjs/toolkit/dist/configureStore';

export type RenderParams = Parameters<typeof render>;

export type RecoilRootPropsCustom = {
  initializeState?: (mutableSnapshot: MutableSnapshot) => void;
  override?: boolean;
};

export interface WrapperOptions {
  routerProps?: MemoryRouterProps;
  storeFactory?: (initStore: any) => ToolkitStore;
  initStore?: any;
  recoilRootProps?: RecoilRootPropsCustom;
  theme?: any;
}

export const createTestWrapper = (options?: WrapperOptions) => {
  const { routerProps, recoilRootProps, theme = lightTheme } = options || {};

  return ({ children }: { children: React.ReactNode }) => {
    const themeComponent = (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    );

    return (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm, hashed: false }}
      >
        <RecoilRoot {...recoilRootProps}>
          <MemoryRouter {...routerProps}>
            <StyledEngineProvider injectFirst>
              {themeComponent}
            </StyledEngineProvider>
          </MemoryRouter>
        </RecoilRoot>
      </ConfigProvider>
    );
  };
};

export const createTesHookWrapper = (options?: WrapperOptions) => {
  const { routerProps, recoilRootProps, theme = lightTheme } = options || {};

  return ({ children }: { children: React.ReactNode }) => {
    const themeComponent = (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    );

    return (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm, hashed: false }}
      >
        <RecoilRoot {...recoilRootProps}>
          <MemoryRouter {...routerProps}>
            <StyledEngineProvider injectFirst>
              {themeComponent}
            </StyledEngineProvider>
          </MemoryRouter>
        </RecoilRoot>
      </ConfigProvider>
    );
  };
};

export const superRender = (
  ...[ui, option, options]: [...RenderParams, WrapperOptions?]
) => {
  const wrapper = createTestWrapper(options);

  return render(ui, {
    wrapper,
    ...option
  });
};

export const superRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHookReact<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  const wrapper = createTesHookWrapper(wrapperOptions);

  return renderHookReact(hook, {
    ...renderHookOptions,
    wrapper
  });
};
