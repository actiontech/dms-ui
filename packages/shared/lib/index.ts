export { default as EmptyBox } from './components/EmptyBox';
export { default as HeaderProgress } from './components/HeaderProgress';
export { default as CronInput } from './components/CronInput';
export { default as BackButton } from './components/BackButton';
export { default as EnterpriseFeatureDisplay } from './components/EnterpriseFeatureDisplay/EnterpriseFeatureDisplay';
export { default as BasicInput } from './components/BasicInput';
export { default as BasicButton } from './components/BasicButton';
export { default as BasicRangePicker } from './components/BasicRangePicker';
export { default as BasicSelect } from './components/BasicSelect';
export { default as PageHeader } from './components/PageHeader';
export { default as BasicSegmented } from './components/BasicSegmented';
export { default as BasicDrawer } from './components/BasicDrawer';
export { default as BasicModal } from './components/BasicModal';
export { default as BasicToolTips } from './components/BasicToolTips';
export { default as BasicInputNumber } from './components/BasicInputNumber';
export { default as BasicSwitch } from './components/BasicSwitch';
export { default as CustomDraggerUpload } from './components/CustomDraggerUpload';
export { default as BasicResult } from './components/BasicResult';
export { default as BasicTable } from './components/BasicTable';
export { default as EditText } from './components/EditText';
export { default as ConfigItem } from './components/ConfigItem';
export { default as BasicInfoList } from './components/BasicInfoList';
export { default as BasicChart } from './components/BasicChart';
export { default as BasicTag } from './components/BasicTag';
export { default as BasicDatePicker } from './components/BasicDatePicker';
export { default as AvatarCom } from './components/AvatarCom';
export { default as DatabaseTypeLogo } from './components/DatabaseTypeLogo';
export { default as BasicTypographyEllipsis } from './components/BasicTypographyEllipsis';
export { default as TokenCom } from './components/TokenCom';
export { default as ToggleTokens } from './components/ToggleTokens';
export { default as SQLRenderer } from './components/SQLRenderer';
export { default as ModeSwitcher } from './components/ModeSwitcher';
export { default as CustomSegmentedFilter } from './components/CustomSegmentedFilter';
export { default as LazyLoadComponent } from './components/LazyLoadComponent';
export { default as SegmentedTabs } from './components/SegmentedTabs';
export { default as SpinIndicator } from './components/SpinIndicator';
export { default as ReminderInformation } from './components/ReminderInformation';
export { default as TestDatabaseConnectButton } from './components/TestDatabaseConnectButton';
export * from './components/SystemConfigurationHub';

export { default as Copy } from './utils/Copy';
export { default as Download } from './utils/Download';
export { default as LocalStorageWrapper } from './utils/LocalStorageWrapper';
export { default as EventEmitter } from './utils/EventEmitter';
export { default as HighlightCode } from './utils/HighlightCode';
export * as FormatterSQL from './utils/FormatterSQL';
export * as Common from './utils/Common';
export * as Math from './utils/Math';
export * as Enum from './enum';
export * as Types from './types/common.type';
