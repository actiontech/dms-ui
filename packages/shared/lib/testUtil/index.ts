export * from './superRender';
export * from './mockHook';
export * from './mockApi';
export * from './customQuery';
export * from './mockRedux';
export * from './common';
export * from './LocationDisplay';
