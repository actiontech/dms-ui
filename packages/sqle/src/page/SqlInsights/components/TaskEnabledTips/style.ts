import { styled } from '@mui/material/styles';

export const LinkTextStyleWrapper = styled('span')`
  cursor: pointer;
  color: ${({ theme }) => theme.sharedTheme.uiToken.colorPrimary} !important;

  &:hover {
    opacity: 0.8;
  }
`;
