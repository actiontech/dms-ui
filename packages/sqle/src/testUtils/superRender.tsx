import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
