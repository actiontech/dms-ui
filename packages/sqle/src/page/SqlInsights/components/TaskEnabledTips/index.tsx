import React from 'react';
import { useTranslation } from 'react-i18next';
import { Space, Typography } from 'antd';
import { LinkTextStyleWrapper } from './style';
import { usePermission, PERMISSIONS } from '@actiontech/shared/lib/features';
import { EmptyBox } from '@actiontech/dms-kit';
export interface TaskEnabledTipsProps {
  onGoToEnable?: () => void;
  instanceId?: string;
}
const TaskEnabledTips: React.FC<TaskEnabledTipsProps> = ({
  onGoToEnable,
  instanceId
}) => {
  const { t } = useTranslation();
  const { checkActionPermission } = usePermission();
  return (
    <Space className="task-enabled-tips">
      <Typography.Text>
        {t('sqlInsights.taskEnabledTips.currentControlTypeNotEnabled')}
      </Typography.Text>
      <EmptyBox
        if={checkActionPermission(
          PERMISSIONS.ACTIONS.SQLE.SQL_MANAGEMENT_CONF.CREATE,
          {
            authDataSourceId: instanceId
          }
        )}
      >
        <LinkTextStyleWrapper onClick={onGoToEnable}>
          {t('sqlInsights.taskEnabledTips.goToEnable')}
        </LinkTextStyleWrapper>
      </EmptyBox>
    </Space>
  );
};
export default TaskEnabledTips;
