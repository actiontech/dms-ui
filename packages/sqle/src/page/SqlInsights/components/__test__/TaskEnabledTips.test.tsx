import { cleanup, screen, fireEvent } from '@testing-library/react';
import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import { mockUsePermission } from '@actiontech/shared/lib/testUtil';
import TaskEnabledTips from '../TaskEnabledTips';

describe('TaskEnabledTips', () => {
  beforeEach(() => {
    mockUsePermission(
      {
        checkActionPermission: jest.fn().mockReturnValue(true)
      },
      {
        useSpyOnMockHooks: true
      }
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('should render component correctly', () => {
    const onGoToEnableMock = jest.fn();
    const { baseElement } = superRender(
      <TaskEnabledTips onGoToEnable={onGoToEnableMock} />
    );

    expect(screen.getByText('当前管控类型未开启')).toBeInTheDocument();
    expect(screen.getByText('[去开启]')).toBeInTheDocument();
    expect(baseElement.querySelector('.task-enabled-tips')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  it('should call onGoToEnable when clicking the enable link', () => {
    const onGoToEnableMock = jest.fn();
    superRender(<TaskEnabledTips onGoToEnable={onGoToEnableMock} />);

    const enableLink = screen.getByText('[去开启]');
    fireEvent.click(enableLink);

    expect(onGoToEnableMock).toHaveBeenCalledTimes(1);
  });

  it('should not render component when checkActionPermission is false', () => {
    mockUsePermission(
      {
        checkActionPermission: jest.fn().mockReturnValue(false)
      },
      {
        useSpyOnMockHooks: true
      }
    );

    const { baseElement } = superRender(
      <TaskEnabledTips onGoToEnable={jest.fn()} />
    );

    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText('当前管控类型未开启')).toBeInTheDocument();
    expect(screen.queryByText('[去开启]')).not.toBeInTheDocument();
  });
});
