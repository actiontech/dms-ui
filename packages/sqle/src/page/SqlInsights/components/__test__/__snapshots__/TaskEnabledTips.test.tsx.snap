// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskEnabledTips should render component correctly 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center task-enabled-tips"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <span
          class="ant-typography"
        >
          当前管控类型未开启，
        </span>
      </div>
      <div
        class="ant-space-item"
      >
        <span
          class="css-12wfj2z"
        >
          去开启
        </span>
      </div>
    </div>
  </div>
</body>
`;
